src/hotspot/share/jfr/jni/jfrJavaSupport.cpp

```diff
@@ -141,7 +141,7 @@ void JfrJavaSupport::notify_all(jobject object, TRAPS) {
   assert(h_obj.not_null(), "invariant");
   ObjectSynchronizer::jni_enter(h_obj, THREAD->as_Java_thread());
   ObjectSynchronizer::notifyall(h_obj, THREAD);
-  ObjectSynchronizer::jni_exit(h_obj(), THREAD);
+  ObjectSynchronizer::jni_exit(THREAD->as_Java_thread(), h_obj());
```

Comment:
For consistency can you switch the parameter order for jni_enter as well in this change?  It looks a little bit odd that jni_exit was changed and not jni_enter.